### DISCUSSION TITLE
How to group indicators on one chart?

### DISCUSSION THREAD

**Discussion by lapitskaya38** [kernc/backtesting.py#1045] [Q&A] [answered]
Hi!
I have four indicators (marked in red "V" in the screenshot) Is it possible to group these indicators on one chart, outside the chart with candles?
![image](https://github.com/kernc/backtesting.py/assets/141924361/627f3a66-fbb4-4bba-9b49-bf0c547b7bcc)

**Comment by peacerider** [comment, marked as answer]
Hi,

Following discussion is may be helpful.
#195 

Example
```
import talib as ta
self.sma1_ = ta.SMA(close, timeperiod=25)
self.sma2_ = ta.SMA(close, timeperiod=45)
self.sma3_ = ta.SMA(close, timeperiod=75)
self.sma1, self.sma2. self.sma3 = self.I(lambda: (self.sma1_, self.sma2_, self.sma3_), name="sma")
```

**Reply by lapitskaya38** [reply]
Thanks a lot.
Have also used this #382 

Perfect:
![image](https://github.com/kernc/backtesting.py/assets/141924361/21f703de-1cf7-4945-80aa-cb085f8170c3)

**Reply by peacerider** [reply]
Hi,

It was very nice.
Until now, I didn't care about legends, but I took this opportunity to try it.

![image](https://github.com/kernc/backtesting.py/assets/99186617/5a69f6b8-65e9-4fa9-b9b6-7dfa6c6d119f)

Succeeded! 
Thank you so much.

`self.ema1, self.ema2 = self.I(my_ema, close, self.n1, self.n2, overlay=False, legends=['ema1', 'ema2'])`

I used the code below.
https://github.com/zlpatel/backtesting.py/commit/a95c555ae006bb40fdbcc9f99c4704bdf5952a4b

If you don't mind, could you tell me the code you used?

**Comment by lapitskaya38** [comment]
`self.M5_ADX_ind, self.M5_MINUS_DI_ind, self.M5_PLUS_DI_ind, self.M5_MA_Tvalue = self.I(lambda: (self.M5_ADX_, self.M5_MINUS_DI_, self.M5_PLUS_DI_, self.M5_MA_), legends=["ADX","MINUS_DI", "PLUS_DI", "Пороговое значение" ], overlay=False)`

**Reply by peacerider** [reply]
Thanks!

**Reply by Rydo111** [reply]
Thanks, very nice. I will definitely use this.

**Comment by Rydo111** [comment]
The cool thing I messed with was dynamic re-scaling of the y-axis. I wanted to plot relative volume, so I created a new plot section, made it a bar chart, added it to the `figs_above_ohlc` list and then used the `CustomJS` callback.

The section code is below the `_plot_volume_section()` function. Code is:

` def _plot_rvol_section():
        """RVOL section"""
        fig = new_indicator_figure(y_axis_label="RVOL")
        fig.height=70
        fig.xaxis.visible = False
        temp = df.copy(deep=True)
        temp = ut.df_flip_caps(df_in=temp, caps=False)
        kwargs = {'fill_method': 'bfill'}
        rvol = rv.ta_rvol(_df=temp, length=50, ma_type='sma', **kwargs)
        rvol = rvol[rvol.columns[0]]
        fig.y_range = DataRange1d(bounds=(0, max(rvol)))
        source.add(rvol.values, name= 'RVOL')
        r = fig.vbar('index', BAR_WIDTH, 'RVOL', source=source, color=inc_cmap)
        set_tooltips(fig, [('RVOL', '@RVOL{0.0}')], renderers=[r])
        fig.yaxis.formatter = NumeralTickFormatter(format="0 a")
        return fig`

The callback code looked like this:
`callback = CustomJS(args={'y_range': fig_rvol.y_range, 'source': source}, code=''`

        clearTimeout(window._autoscale_timeout);
        var Index = source.data.index, rvol = source.data.RVOL, start = cb_obj.start, end = cb_obj.end, min = 0, max = -Infinity;

        for (var i=0; i < Index.length; ++i) {
             if (start <= Index[i] && Index[i] <= end) {
                 max = Math.max(rvol[i], max); min = Math.min(rvol[i], min);
             } 
        } 

        var pad = (max - min) * .09;
        window._autoscale_timeout = setTimeout(function() {
             y_range.start = min - pad; y_range.end = max + pad; 
        });

        '''.replace('\n', '').replace('\t', '')
                        )`

Voila! y-axis now re-sizes dynamically!
